metals/src/main/scala/scala/meta/internal/metals/Trace.scala

```diff
@@ -29,8 +29,17 @@ object Trace {
       AbsolutePath(projectDirectories.cacheDir)
     }.toOption
 
-  val localDirectory: AbsolutePath = PathIO.workingDirectory.resolve(".metals/")
-  val metalsLog: AbsolutePath = localDirectory.resolve("metals.log")
+  private val localDirectory: AbsolutePath =
+    PathIO.workingDirectory.resolve(".metals/")
+
+  // We default to the global cache here mainly because it's not safe to assume
+  // that PathIO.workingDirectory is actually what we want to fallback to.
+  // In editors like VS Code that have a "workspace" concept this will
+  // normally be fine, but in others like nvim where "workspace" doesn't really
+  // exist, we can only rely on the rootUri that is passed in, but we don't have
+  // access to that yet when we use this.
+  val metalsLog: AbsolutePath =
```

Comment:
Just to make sure, will this not cause to always print to the global metals.log? I see that we do set up the logger once we know the workspace, but just wanted to double check if it works as expected.

Reply:
Correct, with my testing it _only_ ever prints like the first 2 log messages that I mentioned before we know the workspace.

Reply:
Ach right, I went to files instead of reading the description properly :face_exhaling: It was a tough day hah

Reply:
> 😮‍💨 It was a tough day hah

Hey, you're doing an amazing job. I can't say that enough. Seriously, keep on keeping on. You inspire me.